priv/templates/phx.gen.live/live_helpers.ex

```diff
@@ -4,6 +4,28 @@ defmodule <%= inspect context.web_module %>.LiveHelpers do
 
   alias Phoenix.LiveView.JS
 
+  @doc """
+  Renders a flash message.
+
+  The rendered flash receives a `:type` that will be used to define
+  proper classes to the element, and a `:message` which will be the
+  inner HTML.
+  If message is empty, the element is not rendered.
+
+  ## Examples
+
+      <.flash type="info" message="User created" />
+  """
+  def flash(assigns) do
+    ~H"""
+    <%= if is_binary(@message) and byte_size(String.trim(@message)) > 0 do %>
```

Comment:
You can move the if outside of HEEx. I also don't think you need to trim the message, as you shouldn't have empty strings as messages. :)